app/lab-integration/lab-sync-pre-processor.service.js

```diff
@@ -0,0 +1,247 @@
+                                      resolve(true);
+                                    } else {
+                                      resolve(false);
+                                    }
+                                  }).catch((error)=> {
+                                      reject('ERROR: hasVisitStartedToday');
+                                  });
+                              }).catch((error)=> {
+                                  reject('ERROR: getPatientVisits');
+                              });
+                               
+                            }
+                      
+                          }).catch((error)=> {
+                            reject('ERROR: Wrong request mode');
+                          });
+
+                     }else{
+                         reject('ERROR: isTestPatient');
+                     }
+
+                }).catch((error) => {
+                    reject('ERROR: isTestPatient');
+                });
+
+      }).catch((error)=> {
+                  reject('ERROR: getPersonAttributes');
+      });
+
+    
+  });
+
+
+}
+
+function getRestResource(path) {
+  var link = config.openmrs.host + ':' + config.openmrs.port + path;
+  if (config.openmrs.https === true) {
+    link = 'https://' + link;
+  } else {
+    link = 'http://' + link;
+  }
+  return link;
+}
+
+function getRequestMode(params) {
+  console.log('getRequestMode', params);
+  return new Promise(function (resolve, reject) {
+    if (params.mode) {
+      const mode = params.mode;
+      if (mode === 'batch') {
+        resolve('batch');
+      }
+    } else {
+      resolve(false);
+    }
+  });
+}
+
+function getPersonAttributes(patientUuid) {
+  var uri = getRestResource('/' + config.openmrs.applicationName + '/ws/rest/v1/person/' + patientUuid);
+  var queryString = {
+    v: 'custom:(attributes:(uuid,display,value,attributeType:(uuid,display))'
+  }
+  // console.log('custom', queryString);
+  return new Promise(function (resolve, reject) {
+    rp.getRequestPromise(queryString, uri)
+      .then(function (response) {
+        resolve(response);
+      })
+      .catch(function (error) {
+        reject(error);
+      })
+  });
+
+}
+
+function isTestPatient(personAttributes) {
+  var testPatientAttributeTypeUuid = '1e38f1ca-4257-4a03-ad5d-f4d972074e69';
+  return new Promise(function (resolve, reject) {
+    var isTestPatient = personAttributes.some((attribute) => {
+      return (attribute.attributeType.uuid === testPatientAttributeTypeUuid) && (attribute.value === true);
+    });
+
+    resolve(isTestPatient);
+
+  });
+
+}
+
+function hasVisitStartedToday(patientVisits) {
+  return new Promise(function (resolve, reject) {
+    var hastodaysVisits = patientVisits.some((visit) => {
+      return (moment(visit.startDatetime).format('YYYY-MM-DD') === moment().format('YYYY-MM-DD'));
+    });
+    resolve(hastodaysVisits);
+
+  });
+
+
+}
+
+function getPatientVisits(patientUuid) {
```

Comment:
This function can be moved to the openmrs resources folder, as it can be re-used later.